code/modules/ruins/russian_invasion.dm

```diff
@@ -0,0 +1,175 @@
+/obj/structure/russian_portal
+	name = "jury-rigged portal"
+	desc = "A damaged alien gateway hooked up to an experimental plasma engine."
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "door"
+	anchored = 1
+	density = 1
+	bound_width = 96
+	bound_height = 96
+	burn_state = LAVA_PROOF
+	luminosity = 1
+	var/points = 0
+
+/obj/structure/russian_portal/attackby(obj/item/I, mob/user, params)
+	..()
+	if(istype(I, /obj/item/stack/sheet/mineral/plasma))
+		var/obj/item/stack/sheet/mineral/plasma/P = I
+		points += P.amount
+		user << "You feed [P.amount] sheets into the machine. It now has [points] sheets of plasma fueling it."
+		qdel(P)
+
+/obj/structure/russian_portal/proc/select_reinforcement(mob/user)
+	var/selection = input(user, "Request Reinforcements", "Portal") as null|anything in list("Prisoner (4)", "Conscript (8)", "Spetsnaz (20)", "Commissar (20)", "Spare Ammo (10)", "Medical Supplies (10)")
+	if(!selection || !Adjacent(user))
+		return
+	switch(selection)
+		if("Prisoner (4)")
```

Comment:
_Gun Hog says something about setting a price to a var, then storing them in a datum before realizing that he is just repeating RR._